src/Domain/TagsTimeDependent.hpp

```diff
@@ -0,0 +1,153 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <boost/optional.hpp>
+#include <cstddef>
+#include <memory>
+#include <string>
+#include <tuple>
+#include <unordered_map>
+
+#include "DataStructures/DataBox/Tag.hpp"
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/EagerMath/Magnitude.hpp"  // For Tags::Normalized
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Domain/CoordinateMaps/CoordinateMap.hpp"
+#include "Domain/FaceNormal.hpp"
+#include "Domain/FunctionsOfTime/FunctionOfTime.hpp"
+#include "Domain/FunctionsOfTime/Tags.hpp"
+#include "Domain/Tags.hpp"
+#include "Time/Tags.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace domain {
+/// \ingroup ComputationalDomainGroup
+/// \brief %Tags for the domain.
+namespace Tags {
+/// The Inertial coordinates, Inertial grid velocity, and the Jacobian and
+/// inverse Jacobian from the Grid to the Inertial frames.
+template <size_t Dim>
+struct CoordinatesVelocityAndJacobians : db::SimpleTag {
+  using type = boost::optional<std::tuple<
+      tnsr::I<DataVector, Dim, Frame::Inertial>,
+      ::InverseJacobian<DataVector, Dim, Frame::Grid, Frame::Inertial>,
+      ::Jacobian<DataVector, Dim, Frame::Grid, Frame::Inertial>,
+      tnsr::I<DataVector, Dim, Frame::Inertial>>>;
+};
+
+/// Computes the Inertial coordinates, Inertial grid velocity, and the Jacobian
+/// and inverse Jacobian from the Grid to the Inertial frames.
+template <typename MapTagGridToInertial>
+struct CoordinatesVelocityAndJacobiansCompute
+    : CoordinatesVelocityAndJacobians<MapTagGridToInertial::dim>,
+      db::ComputeTag {
+  using base = CoordinatesVelocityAndJacobians<MapTagGridToInertial::dim>;
+  static constexpr size_t dim = MapTagGridToInertial::dim;
+
+  using return_type = typename base::type;
+
+  static void function(
+      const gsl::not_null<return_type*> result,
+      const domain::CoordinateMapBase<Frame::Grid, Frame::Inertial, dim>&
+          grid_to_inertial_map,
+      const tnsr::I<DataVector, dim, Frame::Grid>& source_coords,
+      const double time,
+      const std::unordered_map<
+          std::string,
+          std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime>>&
+          functions_of_time) noexcept {
+    // Use identity to signal time-independent
+    if (not grid_to_inertial_map.is_identity()) {
+      *result = grid_to_inertial_map.coords_frame_velocity_jacobians(
+          source_coords, time, functions_of_time);
+    } else {
+      *result = boost::none;
+    }
+  }
+
+  using argument_tags =
+      tmpl::list<MapTagGridToInertial, Tags::Coordinates<dim, Frame::Grid>,
+                 ::Tags::Time, Tags::FunctionsOfTime>;
+};
+
+/// Computes the Inertial coordinates from
+/// `CoordinatesVelocityAndJacobians`
```

Comment:
Could you add a few details what this compute tag does? As far as I understand it either references the Grid coordinates or retrieves the Inertial coords from the other tag, but it's not quite clear from name or docs. Perhaps it would help to name it something like `InertialFromGridCoordinatesCompute`

Reply:
I made the name change. I think we generally don't document how a compute tag computes something, so I'm not clear why we should treat this one differently.